lib/assets/javascripts/manage-prototype/manage-plugins.js

```diff
@@ -23,107 +26,143 @@ window.GOVUKPrototypeKit.documentReady(() => {
   }
 
   const showCompleteStatus = () => {
-    clearTimeout(actionTimeoutId)
+    if (actionTimeoutId) {
+      clearTimeout(actionTimeoutId)
+    }
     log('Successful')
     hide('panel-processing')
     show('panel-complete')
     show('instructions-complete')
   }
 
   const showErrorStatus = () => {
+    if (actionTimeoutId) {
+      clearTimeout(actionTimeoutId)
+    }
     log('Failed')
     hide('panel-processing')
     show('panel-error')
   }
 
-  const postRequest = (url) => fetch(url, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-      'CSRF-Token': token
-    },
-    body: JSON.stringify({
-      package: packageName,
-      version
+  const postRequest = (url) => {
+    const token = document.querySelector('meta[name="csrf-token"]').getAttribute('content')
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'CSRF-Token': token
+      },
+      body: JSON.stringify({
+        package: packageName,
+        version
+      })
     })
-  })
+      .then(response => {
+        return response.json()
+      })
+  }
 
-  const actionTimeoutId = setTimeout(() => {
-    timedOut = true
-  }, timeout)
+  const makeRequest = (fn) => {
+    return new Promise((resolve) => {
```

Comment:
I'm not sure whether we should be polyfilling `Promise` or not.  I'll do some research.

Reply:
I bounced ideas around with @36degrees an I think we should preemptively check whether `Promise` is available, if it's not we don't run any of the JS and they get the non-JS experience.

Reply:
Done as requested